SmartDeviceLinkTests/DevAPISpecs/SDLMenuManagerSpec.m

```diff
@@ -257,17 +262,131 @@ @interface SDLMenuManager()
             });
         });
 
-        describe(@"updating when a menu already exists", ^{
+        describe(@"updating when a menu already exists with dynamic updates on", ^{
             beforeEach(^{
+                testManager.dynamicMenuUpdatesMode = SDLDynamicMenuUpdatesModeForceOn;
+            });
+            
+            it(@"should send deletes first", ^{
                 testManager.menuCells = @[textOnlyCell];
                 [mockConnectionManager respondToLastMultipleRequestsWithSuccess:YES]; // Adds
                 [mockConnectionManager respondToLastMultipleRequestsWithSuccess:YES]; // Submenu
 
                 testManager.menuCells = @[textAndImageCell];
-                [mockConnectionManager respondToLastMultipleRequestsWithSuccess:YES]; // Deletes
+                [mockConnectionManager respondToLastMultipleRequestsWithSuccess:YES];
+
+                NSPredicate *deleteCommandPredicate = [NSPredicate predicateWithFormat:@"self isMemberOfClass:%@", [SDLDeleteCommand class]];
+                NSArray *deletes = [[mockConnectionManager.receivedRequests copy] filteredArrayUsingPredicate:deleteCommandPredicate];
+
+                NSPredicate *addCommandPredicate = [NSPredicate predicateWithFormat:@"self isMemberOfClass:%@", [SDLAddCommand class]];
+                NSArray *adds = [[mockConnectionManager.receivedRequests copy] filteredArrayUsingPredicate:addCommandPredicate];
+
+                expect(deletes).to(haveCount(1));
+                expect(adds).to(haveCount(2));
+            });
+
+            it(@"should send dynamic deletes first then dynamic adds case 2", ^{
```

Comment:
instead of saying "case 2", say something like "with two submenu cells" or whatever description is accurate of this case.

Reply:
fixed